tests/unit/Promise.ts

```diff
@@ -76,19 +70,33 @@ export function addPromiseTests(suite: any, Promise: PromiseType) {
 			}));
 		},
 
-		'sparse array': function () {
-			let dfd = this.async();
-			let iterable: any[] = [];
+		'sparse array': {
+			all() {
+				let dfd = this.async();
+				let iterable: any[] = [];
 
-			iterable[0] = Promise.resolve(0);
-			iterable[3] = Promise.resolve(3);
+				iterable[0] = Promise.resolve(0);
+				iterable[3] = Promise.resolve(3);
 
-			Promise.all(iterable).then(dfd.callback(function (value: number[]) {
-				assert.strictEqual(value[0], 0);
-				assert.isUndefined(value[1]);
-				assert.isUndefined(value[2]);
-				assert.strictEqual(value[3], 3);
-			}));
+				Promise.all(iterable).then(dfd.callback(function (value: number[]) {
+					assert.strictEqual(value[0], 0);
+					assert.isUndefined(value[1]);
+					assert.isUndefined(value[2]);
+					assert.strictEqual(value[3], 3);
+				}));
+			},
+
+			race() {
+				let dfd = this.async();
+				let iterable: any[] = [];
+
+				iterable[0] = Promise.resolve(0);
+				iterable[3] = Promise.resolve(3);
+
+				Promise.race(iterable).then(dfd.callback(function (value: number) {
+					assert.strictEqual(value, 0);
```

Comment:
Yeah, it would make more sense to set index 1 instead of 0.

I would assume a `race` that didn't handle sparse arrays correctly would simply not skip unset indices. It would access indices 1 and 2 in the current test and treat the resulting `undefineds` as meaningful values. In the current test that shouldn't matter since the first index is set. If we set index 1 instead of 0, though, then a bad `race` would resolve with `undefined`.